Sources/FoundationExtensions/Promise/PromiseType+Init.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  PromiseType+Init.swift
+//  FoundationExtensions
+//
+//  Created by Luiz Barbosa on 11.01.19.
+//  Copyright © 2020 Lautsprecher Teufel GmbH. All rights reserved.
+//
+
+#if canImport(Combine)
+import Combine
+import Foundation
+
+@available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+extension PromiseType {
+    /// A promise from a hardcoded successful value
+    /// - Parameter value: a hardcoded successful value. It's gonna be evaluated on demand from downstream
+    public init(value: Success) {
+        self.init { Just(value).mapError(absurd).eraseToAnyPublisher() }
+    }
+
+    /// A promise from a hardcoded error
+    /// - Parameter error: a hardcoded error. It's gonna be evaluated on demand from downstream
+    public init(error: Failure) {
+        self.init { Fail(error: error).map(absurd).eraseToAnyPublisher() }
+    }
+
+    /// A promise from a hardcoded result value
+    /// - Parameter value: a hardcoded result value. It's gonna be evaluated on demand from downstream
+    public init(result: Result<Success, Failure>) {
+        self.init { result.publisher.eraseToAnyPublisher() }
+    }
+
+    /// Creates a new promise by evaluating a synchronous throwing closure, capturing the
+    /// returned value as a success, or any thrown error as a failure.
+    ///
+    /// - Parameters:
+    ///   - body: A throwing closure to evaluate.
+    ///   - errorTransform: a way to transform the throwing error from type `Error` to type `Failure` of this `PromiseType`
+    @_transparent
```

Comment:
Okay, this was weird to me, so I googled it and basically the only information about this I can find is that it inlines and that you should not use it in a real world project.

https://swiftrocks.com/the-forbidden-inline-attribute-in-swift.html

> Since @inline is not officially supported, you should really never use it in a real project and use this article only for the sake of learning something new.

Do you disagree on this?

Reply:
This shouldn't be here. This was introduced in the old Promises and I don't remember exactly why, but I think it was to solve some compiler over-optimisation causing the app to crash in runtime. Probably this doesn't make sense any more neither for the Combine Promises implementation nor for the new compiler versions.

Removed now.